terraform/modules/single_instance/cloudinit.tf

```diff
@@ -0,0 +1,84 @@
+data "cloudinit_config" "main" {
+  gzip          = false
+  base64_encode = false
+  part {
+    filename     = "cloud-config.yaml"
+    content_type = "text/cloud-config"
+
+    content = templatefile("${local.scripts_dir}/cloud-config.yaml", { files = data.local_file.ansible })
+  }
+  part {
+    filename     = "userscript.sh"
+    content_type = "text/x-shellscript"
+    content = var.userscript
+  }
+  dynamic "part" {
+    for_each = var.cloudinit
+    content {
+      filename = part.key
+      content_type = part.value.content_type
+      content = part.value.content
+    }
+  }
+}
+locals {
+  ansible_files = fileset("${local.scripts_dir}/ansible/", "**")
+}
+data "local_file" "ansible" {
+  for_each = local.ansible_files
+  filename = "${local.scripts_dir}/ansible/${each.value}"
+}
+
+resource "time_sleep" "waitforinstall" {
+  depends_on = [ openstack_compute_instance_v2.instance_01 ]
+  create_duration = "2m"
```

Comment:
did you check if this sleep time of 2 minutes is enough for all the cases? what happens if there is a timeout?